config/haproxy.cfg

```diff
@@ -1,5 +1,9 @@
 template {
   source = "/consul-template/template.d/haproxy.tmpl"
   destination = "/haproxy/haproxy.cfg"
-  command = "/usr/sbin/haproxy -D -p /var/run/haproxy.pid  -f /haproxy/haproxy.cfg -sf $(cat /var/run/haproxy.pid) || true"
+}
+
+exec {
+  reload_signal = "SIGUSR2"
+  command = "/usr/sbin/haproxy -W -p /var/run/haproxy.pid -f /haproxy/haproxy.cfg"
```

Comment:
This is a new mode in haproxy 1.8 called master-worker:
https://cbonte.github.io/haproxy-dconv/1.8/management.html#4

The process tree will look like this:

consul-template
-> haproxy (master)
  -> haproxy (worker)

When consul-template renders a new template, it will send SIGUSR2 to haproxy (master), which will start a new haproxy (worker) and terminate the old one.

If haproxy dies, then consul-template dies, and then the container dies and gets restarted by docker.